docs/rfcs/016-user-and-database-management.md

```diff
@@ -0,0 +1,68 @@
+## Zentih user and database management RFC
+
+### CREATE USER
+
+CREATE USER is only allowed from console, 
+because it requires generation of the access token and other complex logic.
+
+Client has to chose which databases will be accessible by the new user.
+Along with database access, we explicitly grant CREATE ON DATABASE privilege to the user.
+`GRANT CREATE ON DATABASE .. TO .. ;`
+to let the user to create extensions.
+
+All new users are granted with CREATEDB privilege by default.
+`CREATE ROLE .. WITH CREATEDB`
+Later we may add an option to manage it.
+
+*Q: Are we going to allow user to create new users and NOLIGIN roles to manage privileges?
+*`CREATE ROLE .. WITH CREATEROLE`
+
+
+### CREATE DATABASE
+
+Client can CREATE DATABASE via SQL as well as via console.
+
+When database is created from SQL (not by superuser)
+we intercept this action using DDL hook and send it to the console to synchronize the information about databases.
+
+### CREATE EXTENSION
+
+To create trusted extensions in the database, user must have an access privileges on the database.
+
+We explicitly grant CREATE ON DATABASE privilege to the new user along with database access.
+If user creates a database, he becomes the owner and thus has CREATE ON DATABASE privilege by default.
```

Comment:
Is it really true? We already set default user as an owner of default database `main`, but it cannot create extensions

Reply:
It will become true after https://github.com/zenithdb/zenith/pull/1477